osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -184,13 +177,90 @@ public ChatOverlay()
                 else
                     textbox.HoldFocus = true;
             };
+            channelSelection.OnRequestJoin = channel =>
+            {
+                if (!channelManager.JoinedChannels.Contains(channel))
+                    channelManager.JoinedChannels.Add(channel);
+            };
+            channelSelection.OnRequestLeave = channel => channelManager.JoinedChannels.Remove(channel);
+        }
+
+        private void availableChannelsChanged(object sender, NotifyCollectionChangedEventArgs args)
+        {
+            channelSelection.Sections = new[]
+            {
+                new ChannelSection
```

Comment:
This seems like the wrong place to do this. If anything, `AvailableChannels` should be passed in to `channelSelection` and the drawable construction should happen there.